base/libgit2/oid.jl

```diff
@@ -155,17 +155,15 @@ function GitShortHash(obj::GitObject)
     return sid
 end
 
-Base.hex(id::GitHash) = join([hex(i,2) for i in id.val])
-Base.hex(id::GitShortHash) = hex(id.hash)[1:id.len]
-
 """
     raw(id::GitHash) -> Vector{UInt8}
 
 Obtain the raw bytes of the [`GitHash`](@ref) as a vector of length $OID_RAWSZ.
 """
 raw(id::GitHash) = collect(id.val)
 
-Base.string(id::AbstractGitHash) = hex(id)
+Base.string(id::GitHash) = join([string(i, base = 16, pad = 2) for i in id.val])
+Base.string(id::GitShortHash) = string(id.hash)[1:id.len]
```

Comment:
I think the last discussion was to define `Base.print(io::IO, id::GitHash)` (which implicitly defines `Base.string`).

Reply:
Oh no was it I'm so confused

Reply:
e.g.
```
function Base.print(io::IO, id::GitHash)
    for i in id.val
        print(io, string(i, base = 16, pad = 2))
    end
end
```

Reply:
Though this makes me think that we should also have the same args to `print` as well...

Reply:
perhaps best to leave that for another PR though